test/cppsuite/tests/cursor_bound_01.cpp

```diff
@@ -0,0 +1,695 @@
+     * bounds on the range cursor. The lower and upper bounds are randomly generated strings and the
+     * inclusive configuration is also randomly set as well.
+     */
+    std::pair<bound, bound>
+    set_random_bounds(thread_context *tc, scoped_cursor &range_cursor)
+    {
+        int ret;
+        bound lower_bound, upper_bound;
+
+        auto set_random_bounds = random_generator::instance().generate_integer(0, 3);
+        if (set_random_bounds == NO_BOUNDS)
+            range_cursor->bound(range_cursor.get(), "action=clear");
+
+        if (set_random_bounds == LOWER_BOUND_SET || set_random_bounds == ALL_BOUNDS_SET) {
+            /* Reverse case. */
+            if (_reverse_collator_enabled)
+                lower_bound = bound(tc->key_size, true, std::string(tc->key_size, 'z'));
+            /* Normal case. */
+            else
+                lower_bound = bound(tc->key_size, true, "0");
+            range_cursor->set_key(range_cursor.get(), lower_bound.get_key().c_str());
+            ret = range_cursor->bound(range_cursor.get(), lower_bound.get_config().c_str());
+            testutil_assert(ret == 0 || ret == EINVAL);
+        }
+
+        if (set_random_bounds == UPPER_BOUND_SET || set_random_bounds == ALL_BOUNDS_SET) {
+
+            /* Reverse case. */
+            if (_reverse_collator_enabled)
+                upper_bound = bound(tc->key_size, false, "0");
+            /* Normal case. */
+            else
+                upper_bound = bound(tc->key_size, false, std::string(tc->key_size, 'z'));
+            range_cursor->set_key(range_cursor.get(), upper_bound.get_key().c_str());
+            ret = range_cursor->bound(range_cursor.get(), upper_bound.get_config().c_str());
+            testutil_assert(ret == 0 || ret == EINVAL);
+        }
+
+        return std::make_pair(lower_bound, upper_bound);
+    }
+
+    /*
+     * Validate the bound search_near call. There are three scenarios that needs to be validated
+     * differently.
+     *  Scenario 1: Range cursor has returned WT_NOTFOUND, this indicates that no records exist in
+     * the bounded range. Validate this through traversing all records within the range on a normal
+     * cursor.
+     *  Scenario 2: Range cursor has returned a key and the search key is outside the range bounds.
+     * Validate that the returned key is either the first or last record in the bounds.
+     *  Scenario 3: Range cursor has returned a key and the search key is inside the range bounds.
+     * Validate that the returned key is visible and that it is indeed the closest key that range
+     * cursor could find.
+     */
+    void
+    validate_bound_search_near(int range_ret, int range_exact, scoped_cursor &range_cursor,
+      scoped_cursor &normal_cursor, const std::string &search_key, const bound &lower_bound,
+      const bound &upper_bound)
+    {
+        /* Range cursor has successfully returned with a key. */
+        if (range_ret == 0) {
+            const char *key;
```

Comment:
np: Sorry, I know you added the other variables in a recent commit but it pushed this one away from where it is used line 278.